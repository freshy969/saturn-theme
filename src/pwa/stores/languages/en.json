{
  "powered": "powered by",
  "classic": "Classic Version",
  "loadMore": "Load more",
  "writtenBy": "Written by",
  "shared": "#1# shared",
  "sharedPlural": "#3# shared",
  "readingTime": "#1# minute",
  "readingTimePlural": "#1# minutes",
  "next": "Next",
  "loading": "Loading...",
  "share": "Share",
  "copyLink": "Copy link",
  "copied": "Copied",
  "nextPosts": "Next posts",
  "moreInCategory": "More in #category#",
  "notifications": "Notifications"
}
