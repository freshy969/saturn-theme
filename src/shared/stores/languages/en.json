{
  "poweredBy": "powered by",
  "classicVersion": "Classic Version",
  "loadMore": "Load more",
  "writtenBy": "Written by",
  "shares": "Share",
  "sharesPlural": "Shares",
  "sharesWithNumber": "#number# share",
  "sharesWithNumberPlural": "#number# shares",
  "readingTime": "#number# minute",
  "readingTimePlural": "#number# minutes",
  "next": "Next",
  "loading": "Loading...",
  "share": "Share",
  "copyLink": "Copy link",
  "copiedLink": "Copied",
  "interestedPosts": "You might be interested in...",
  "nextPosts": "Next posts",
  "moreInCategory": "More in #category#",
  "notifications": "Notifications"
}
