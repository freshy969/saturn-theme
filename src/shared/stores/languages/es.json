{
  "poweredBy": "powered by",
  "classicVersion": "Versión clásica",
  "loadMore": "Cargar más",
  "writtenBy": "Escrito por",
  "shares": "Compartido",
  "sharesPlural": "Compartidos",
  "sharesWithNumber": "#number# compartido",
  "sharesWithNumberPlural": "#number# compartidos",
  "readingTime": "#number# minuto",
  "readingTimePlural": "#number# minutos",
  "next": "Siguiente",
  "loading": "Cargando...",
  "share": "Compartir",
  "copyLink": "Copiar link",
  "copiedLink": "Copiado",
  "interestedPosts": "Te puede interesar...",
  "nextPosts": "Siguientes artículos",
  "moreInCategory": "Más en #category#",
  "notifications": "Notificaciones"
}
