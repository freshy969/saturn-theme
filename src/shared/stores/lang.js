import { types } from 'mobx-state-tree';
import * as languages from './languages';

export default types
  .model('Localization')
  .props({
    default: types.optional(types.frozen, languages.en),
    current: types.optional(types.frozen, {}),
  })
  .views(self => ({
    get(key) {
      return self.current[key] || self.default[key];
    },
    getReadingTime(time) {
      const text = time === 1 ? self.get('readingTime') : self.get('readingTimePlural');
      return text.replace('#number#', time);
    },
    getShares(count) {
      return count === 1 ? self.get('shares') : self.get('sharesPlural');
    },
    getSharesWithNumber(count) {
      const text = count === 1 ? self.get('sharesWithNumber') : self.get('sharesWithNumberPlural');
      return text.replace('#number#', count);
    },
    getMoreInCategory(category) {
      return self.get('moreInCategory').reclate('#category#', category);
    },
  }))
  .actions(self => ({
    setLang(lang) {
      self.current = languages[lang] || {};
    },
  }));
