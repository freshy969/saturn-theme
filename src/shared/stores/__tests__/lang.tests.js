import lang from '../lang';
import * as languages from '../languages';

describe('Theme › Shared › Stores › Lang', () => {
  test('store has `default`, `current` and `code` props', () => {
    const self = lang.create();

    expect(self.default).not.toBe(undefined);
    expect(self.current).not.toBe(undefined);
    expect(self.code).not.toBe(undefined);
    expect(self).toMatchSnapshot();
  });

  test('`default` language should be English', () => {
    const self = lang.create();

    expect(self.default).toEqual(languages.en);
  });

  test('default language `code` should be English', () => {
    const self = lang.create();

    expect(self.code).toBe('en');
  });

  test('`setLang` should change the value of `current` and `code` to Spanish', () => {
    const self = lang.create();

    expect(self.current).toEqual({});
    expect(self.code).toBe('en');

    self.setLang('es');

    expect(self.current).toEqual(languages.es);
    expect(self.code).toBe('es');
    expect(self).toMatchSnapshot();
  });

  test('`setLang` should change the value of `current` to an empty object if the code provided is not available', () => {
    const self = lang.create({
      current: languages.es,
      code: 'es',
    });

    expect(self.current).toEqual(languages.es);
    expect(self.code).toBe('es');

    self.setLang('ar');

    expect(self.current).toEqual({});
    expect(self.code).toBe('ar');
    expect(self).toMatchSnapshot();
  });

  test('`get` retrieves the string from the `default` language when that string is not available in `current`', () => {
    const self = lang.create();
    const str = self.get('loadMore');

    expect(str).toBe('Load more');
  });

  test('`get` retrieves the string from the `current` language if available', () => {
    const self = lang.create();
    self.setLang('es');
    const str = self.get('loadMore');

    expect(str).toBe('Cargar más');
  });

  test('`getReadingTime` returns a string in singular with the time provided', () => {
    const self = lang.create();
    const str = self.getReadingTime(1);

    expect(str).toBe('1 minute');
  });

  test('`getReadingTime` returns a string in plural with the time provided', () => {
    const self = lang.create();
    const str = self.getReadingTime(10);

    expect(str).toBe('10 minutes');
  });

  test('`getShares` returns a string in singular', () => {
    const self = lang.create();
    const str = self.getShares(1);

    expect(str).toBe('Share');
  });

  test('`getShares` returns a string in plural', () => {
    const self = lang.create();
    const str = self.getShares(2);

    expect(str).toBe('Shares');
  });

  test('`getSharesWithNumber` returns a string in singular with the number provided', () => {
    const self = lang.create();
    const str = self.getSharesWithNumber(1);

    expect(str).toBe('1 share');
  });

  test('`getSharesWithNumber` returns a string in plural with the number provided', () => {
    const self = lang.create();
    const str = self.getSharesWithNumber(5);

    expect(str).toBe('5 shares');
  });

  test('`getMoreInCategory` returns a string in with the category provided', () => {
    const self = lang.create();
    const str = self.getMoreInCategory('Pachachos');

    expect(str).toBe('More in Pachachos');
  });
});
