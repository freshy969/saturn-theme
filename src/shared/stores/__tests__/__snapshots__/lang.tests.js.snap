// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme › Shared › Stores › Lang \`setLang\` should change the value of \`current\` and \`code\` to Spanish 1`] = `
Object {
  "code": "es",
  "current": Object {
    "classicVersion": "Versión clásica",
    "copiedLink": "Copiado",
    "copyLink": "Copiar link",
    "interestedPosts": "Te puede interesar...",
    "loadMore": "Cargar más",
    "loading": "Cargando...",
    "moreInCategory": "Más en #category#",
    "next": "Siguiente",
    "nextPosts": "Siguientes artículos",
    "notifications": "Notificaciones",
    "poweredBy": "powered by",
    "readingTime": "#number# minuto",
    "readingTimePlural": "#number# minutos",
    "share": "Compartir",
    "shares": "Compartido",
    "sharesPlural": "Compartidos",
    "sharesWithNumber": "#number# compartido",
    "sharesWithNumberPlural": "#number# compartidos",
    "writtenBy": "Escrito por",
  },
  "default": Object {
    "classicVersion": "Classic Version",
    "copiedLink": "Copied",
    "copyLink": "Copy link",
    "interestedPosts": "You might be interested in...",
    "loadMore": "Load more",
    "loading": "Loading...",
    "moreInCategory": "More in #category#",
    "next": "Next",
    "nextPosts": "Next posts",
    "notifications": "Notifications",
    "poweredBy": "powered by",
    "readingTime": "#number# minute",
    "readingTimePlural": "#number# minutes",
    "share": "Share",
    "shares": "Share",
    "sharesPlural": "Shares",
    "sharesWithNumber": "#number# share",
    "sharesWithNumberPlural": "#number# shares",
    "writtenBy": "Written by",
  },
}
`;

exports[`Theme › Shared › Stores › Lang \`setLang\` should change the value of \`current\` to an empty object if the code provided is not available 1`] = `
Object {
  "code": "ar",
  "current": Object {},
  "default": Object {
    "classicVersion": "Classic Version",
    "copiedLink": "Copied",
    "copyLink": "Copy link",
    "interestedPosts": "You might be interested in...",
    "loadMore": "Load more",
    "loading": "Loading...",
    "moreInCategory": "More in #category#",
    "next": "Next",
    "nextPosts": "Next posts",
    "notifications": "Notifications",
    "poweredBy": "powered by",
    "readingTime": "#number# minute",
    "readingTimePlural": "#number# minutes",
    "share": "Share",
    "shares": "Share",
    "sharesPlural": "Shares",
    "sharesWithNumber": "#number# share",
    "sharesWithNumberPlural": "#number# shares",
    "writtenBy": "Written by",
  },
}
`;

exports[`Theme › Shared › Stores › Lang store has \`default\`, \`current\` and \`code\` props 1`] = `
Object {
  "code": "en",
  "current": Object {},
  "default": Object {
    "classicVersion": "Classic Version",
    "copiedLink": "Copied",
    "copyLink": "Copy link",
    "interestedPosts": "You might be interested in...",
    "loadMore": "Load more",
    "loading": "Loading...",
    "moreInCategory": "More in #category#",
    "next": "Next",
    "nextPosts": "Next posts",
    "notifications": "Notifications",
    "poweredBy": "powered by",
    "readingTime": "#number# minute",
    "readingTimePlural": "#number# minutes",
    "share": "Share",
    "shares": "Share",
    "sharesPlural": "Shares",
    "sharesWithNumber": "#number# share",
    "sharesWithNumberPlural": "#number# shares",
    "writtenBy": "Written by",
  },
}
`;
